control-plane/tests/test_cli_inspection.py

```
@@ -589,6 +590,272 @@ def __init__(self, server_address: tuple[str, int], handler_class: type) -> None
+                for _ in range(100):
+                    if servers:
+                        break
+                    thread.join(0.01)
+                self.assertTrue(servers, "expected test HTTP server to start")
+
+                base_url = f"http://127.0.0.1:{servers[0].server_port}"
+                diagnostics_payload = json.loads(
+                    request.urlopen(
+                        f"{base_url}/diagnostics/readiness",
+                        timeout=2,
+                    ).read().decode("utf-8")
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)